### DISCUSSION TITLE
the graph and its abscissa

### DISCUSSION THREAD

**Discussion by MoucheTaMoustache** [webspiderteam/Bluetooth-DMM-For-Windows#26] [Ideas]
Hello,
First of all, thank you very much for your program! He is fantastic. You have realized what I have been struggling to do for several months!

I had a suggestion concerning the abscissa of the graph. The abscissa is time. Is it possible to change the timeline to date-time (dd/mm/yy - hh:mm:ss:ms) or create a chart option to customize the abscissa line?
This would allow me to have continuous measurements of the temperature and to identify the exact moment of its variations.

So when exporting to CSV, I will study the raw data and the exact moment of their variation (h/m/s).

Thanks again, that's great!
Moustache

**Comment by webspiderteam** [comment]
Thank you too. Hearig this makes me happy. Thats what I wanted.

All of them posibble but all of them can make issues.
Let me explain


> change the timeline to date-time (dd/mm/yy - hh:mm:ss:ms)

Its possible but cant be milisecond because chart refreshes 1 per second. other than this, there is 2 options 
1. I can change only view format. But If I make this it may not show exact time because if device in AUTO or NFC mode chart does not updates. Also if there is connection loss chart will not update. that means chart wil have lost times and will be different between exact time.
2. I can add date time to data. But if I add it main data takes 4 bytes for each second.   That means it will take 14.4Kb in an Hour. If I add date-time (dd/mm/yy - hh:mm:ss) it add 18bytes per second so that means 79.2 Kb per hour. These values are raw only while its in memory it makes 5 times more memory and also slows down the app also can couse memory leak issues, frezing app and can crash.  

There is another option i can do. There is a raw data logging but it only gives exact data which came from device with out decoding to values, and this log only updates when data changed. But I can make another option for logging with date-time and datas and also this can give you miliseconds too. Again it will update when value changed. This can work but it can make delay to update display.

**Comment by ludwich66** [comment]
Hi MoucheTaMoustache,

Did you see the MQTT Data-Stream Option?
Here are more Parameters Configurable.
Did you read WIKI under https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/wiki?

Ludwich

**Comment by MoucheTaMoustache** [comment]
H‌i webspiderteam!

Thank you for your comprehensive answer!
I didn't know that millisecond will lag software, risking it to crash. I tried to do something like that with python. I used to tried something like with Python that but I had to stop because I don't have the necessary knowledge in coding... XD.

If I understand, it's a matter of balance between program's ability to process information and memory's ability to store it.
In regard of this, milliseconds would be excessives.

Thank you again for your work!
MoucheTaMoustache
 

De : "webspiderteam"
A : "webspiderteam/Bluetooth-DMM-For-Windows" ,"MoucheTaMoustache" ,"Author"
Envoyé: mercredi 24 Août 2022 14:16
Objet : Re: [webspiderteam/Bluetooth-DMM-For-Windows] the graph and its abscissa (Discussion #26)
 

 

Thank you too. Hearig this makes me happy. Thats what I wanted.

All of them posibble but all of them can make issues.
Let me explain


change the timeline to date-time (dd/mm/yy - hh:mm:ss:ms)


Its possible but cant be milisecond because chart refreshes 1 per second. other than this, there is 2 options


	I can change only view format. But If I make this it may not show exact time because if device in AUTO or NFC mode chart does not updates. Also if there is connection loss chart will not update. that means chart wil have lost times and will be different between exact time.
	I can add date time to data. But if I add it main data takes 4 bytes for each second. That means it will take 14.4Kb in an Hour. If I add date-time (dd/mm/yy - hh:mm:ss) it add 18bytes per second so that means 79.2 Kb per hour. These values are raw only while its in memory it makes 5 times more memory and also slows down the app also can couse memory leak issues, frezing app and can crash.


There is another option i can do. There is a raw data logging but it only gives exact data which came from device with out decoding to values, and this log only updates when data changed. But I can make another option for logging with date-time and datas and also this can give you miliseconds too. Again it will update when value changed. This can work but it can make delay to update display.

**Reply by webspiderteam** [reply]
> I didn't know that millisecond will lag software, risking it to crash

I can do all but depends of code and pc. Every writing action takes time if your HDD response slow it can take time much more I can handle it after displaying value but does it really need in miliseconds? because long time measuring does not need that frequent most of times. ofcourse there is extreme conditions can be. 

I want to help you. If you explain where do you want to use and what for. then I think I can find a solution for you.

**Reply by MoucheTaMoustache** [reply]
> I want to help you. If you explain where do you want to use and what for. then I think I can find a solution for you.

Hi @webspiderteam  !
Thank for your help. 

I bought the device (ANENG-9002) for a project of continuous measurements of the resistance of an ionic solution. I need to collect the resistance variation over time: over a period of 30 minutes to 24 hours depending on the tests. I need each resistances and their times (date + hours and minutes at least and I admit that I don't need milliseconds). Then I want to extract the resistances and their time in a .csv file to analyze them.
I don't know what is possible to do, either extract everything after each try (which is currently the case), or have the program write directly to a .csv file over time to relieve its memory.

I don't want to abuse your time.
Thank you, have a nice sunday!

I will read wikipedia's page.

**Reply by webspiderteam** [reply]
Then I have 2 solutions for you. 
One of them is using mqtt without changing app: I made a test server and client for Mqtt and you can get what ever value you want that app has and also there is time and date in every seconds. and I can easyly turn it in to a logger. 

![image](https://user-images.githubusercontent.com/85828505/187076094-260bc1c0-e4c2-4a5a-9cb8-0862d5031b63.png)

https://github.com/webspiderteam/Bluetooth_DMM_For_Windows_MQTT_Test_Project 

Other one is adding log file vhen value changes: this one won't log all datas but chan resistans changed you will see also there wont be a lot of data it will just show you when it changed and it can have date and time values even miliseconds too. but of course some dummy  changes too because sometimes values goes up and down because of ability of Multimeter.

so tell me which one is better for you?

**Reply by MoucheTaMoustache** [reply]
Hi!
Thank you for your rapid reply ! The first option is the best to my project.
What have I to do to obtain this?
Thank you again!

**Reply by webspiderteam** [reply]
just tell me which values do you need. then I will adopt test app to your needs. and then give me a few hours then I will send you the app brobably 2 or 3 hour later

**Comment by MoucheTaMoustache** [comment]
@webspiderteam you are a warrior! 

Values that I would have need : 
 - resistance (ohm)
 - temperature (°C)
 - time & date

There is no emergency! Enjoy your sunday!
MoucheTaMoustache

**Reply by webspiderteam** [reply]
Are there more than 1 device?


> There is no emergency! Enjoy your sunday!

n.p I have time now

**Reply by webspiderteam** [reply]
[Release.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/9440124/Release.zip) Use this app 

Default settings are  like that just check Create server and connect before start Bluetooth dmm
![image](https://user-images.githubusercontent.com/85828505/187086279-9791105c-d19b-4c06-9386-beffedbece08.png)

Start Bluetooth DMM and go to settings. In settings make MQTT Client enabled and enter MQTT Settings. In MQTT Settings leave settings as it is. Like picture below.
![image](https://user-images.githubusercontent.com/85828505/187086486-3593e24a-6663-4ae7-9e0d-a4261833bcec.png)

Enter Values settings and Check Selected Values. Default values Should be like that. Just be sure.
![image](https://user-images.githubusercontent.com/85828505/187086505-2966ea29-b50e-46ff-be23-61936d017aa2.png)

then click save on each window. now you can see values will filled in list when you connected Multimeter to app.
Then Click Start Logging whenever you want. Thats it. When you done click Stop Logging.

there is two things you have to careful. first if you change any setting both of app must have same settings. Second thing is you need to start Mqtttest app before start the Bluetooth DMM because if you dont Bluetooth dmm  wont connect until entering setting and save.

**Comment by MoucheTaMoustache** [comment]
@‌webspiderteam
No no, I have one device.
 

De : "webspiderteam"
A : "webspiderteam/Bluetooth-DMM-For-Windows" ,"MoucheTaMoustache" ,"Author"
Envoyé: dimanche 28 Août 2022 16:46
Objet : Re: [webspiderteam/Bluetooth-DMM-For-Windows] the graph and its abscissa (Discussion #26)
 

 

Are there more than 1 device?

**Comment by MoucheTaMoustache** [comment]
T‌hank you very much!
I am at my job, I can't try it until this evening. I'll keep you informed!
Have a nice day!
 
 

De : "webspiderteam"
A : "webspiderteam/Bluetooth-DMM-For-Windows" ,"MoucheTaMoustache" ,"Author"
Envoyé: dimanche 28 Août 2022 19:37
Objet : Re: [webspiderteam/Bluetooth-DMM-For-Windows] the graph and its abscissa (Discussion #26)
 

 

Release.zip Use this app

Default settings are like that just check Create server and connect before start Bluetooth dmm


Start Bluetooth DMM and go to settings. In settings make MQTT Client enabled and enter MQTT Settings. In MQTT Settings leave settings as it is. Like picture below.


Enter Values settings and Check Selected Values. Default values Should be like that. Just be sure.


then click save on each window. now you can see values will filled in list when you connected Multimeter to app.
Then Click Start Logging whenever you want. Thats it. When you done click Stop Logging.

there is two things you have to carryful. first if you change any setting both of app must have same settings. Second thing is you need to start Mqtttest app before start the Bluetooth DMM because if you dont Bluetooth dmm wont connect until entering setting and save.

**Reply by webspiderteam** [reply]
Thank you. I will wait your results 🙂

**Reply by MoucheTaMoustache** [reply]
Hi!
I have tried this morning! It works perfectly! 

![capture](https://user-images.githubusercontent.com/107130427/187621861-87abc8e4-4f21-44f6-a077-839a2d9a627c.jpg)
I captured my screen : to the left the export data from _Bluetooth DMM app_; to the right the _mqtttest_.

How to export data directly from _mqtttest_? I tried to copy/paste, select all text from. I think I can do it manually in excel export file.
If time 0 = 31/8/22 ; 9h31:4**7** then time 1 = 31/8/22 ; 9h31:4**8** and so on.

However, waow, I am so happy! Thank you very very much!

**Reply by webspiderteam** [reply]
This log from Bluetooth DMM. Mqtt test apps log is under mqtt test apps folder.

**Reply by webspiderteam** [reply]
oh I forgot to change file extention from txt to csv sorry.
![image](https://user-images.githubusercontent.com/85828505/187626521-e3cb1ee0-f78c-4dc7-a4dc-3689712439b5.png)

if you change log txt to log.csv you can see it like that
![image](https://user-images.githubusercontent.com/85828505/187626912-b828c521-7c08-4a1b-922e-7f0a89637ff6.png)

**Reply by MoucheTaMoustache** [reply]
> This log from Bluetooth DMM. Mqtt test apps log is under mqtt test apps folder.

Oh I'm sorry, I didn't see the .txt doc!

**Comment by webspiderteam** [comment]
I fixed extention and also that weird charcter problem when its degree.
[Release.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/9460004/Release.zip)

**Reply by MoucheTaMoustache** [reply]
That's very very nice! Thank you very much!
Honestly, I tried to do it with Python like I said above. I tried to personalise Rick's work (here: [https://github.com/riktw/AN9002_info]). I needed help from Python community. However, the french community I found wasn't helpfull and pretty disparaging. I have abandoned my project for another time with cynical.
I knocked here by chance, so happy to find your work.

Frankly I am flabbergasted by your kindness! Thank you again!
Sincerely yours!
MTM

**Reply by webspiderteam** [reply]
Thank you for your nice thoughts. We are all human and we have to be kind each other. I know tere are people around like you told. But they forgot one upon a time they were like us. BTW if you wish I can help for personalize. I know pyton too but now too much but I think I can help as much as I can. If you need and if you wish ofcourse.

**Comment by MoucheTaMoustache** [comment]
H‌i Ludwich, 
thank you for your answer. I didn't see this. I don't know what is it! ^^
How to find that?

So, no I didn't red wiki, I will read it.
Thank you !
MoucheTaMoustache
 

De : "ludwich66"
A : "webspiderteam/Bluetooth-DMM-For-Windows" ,"MoucheTaMoustache" ,"Author"
Envoyé: mercredi 24 Août 2022 15:45
Objet : Re: [webspiderteam/Bluetooth-DMM-For-Windows] the graph and its abscissa (Discussion #26)
 

 

Hi MoucheTaMoustache,

Did you see the MQTT Data-Stream Option?
Here are more Parameters Configurable.
Did you reed https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/wiki?

Ludwich

**Reply by ludwich66** [reply]
Le wiki a été complètement mis à jour depuis la publication - pouvez-vous tout trouver ? Peut-on encore apporter des améliorations ?
salutations de Munich
Ludwich
